### ISSUE TITLE
Module Outputs

### ISSUE THREAD

**Issue by eddytrex** [hashicorp/terraform#22502]
### Terraform Version

```
terraform version
Terraform v0.12.5
+ provider.null v2.1.2
```

### Terraform Configuration Files

module 1:
```hcl
provider "null" {}

resource "null_resource" "name" {
  count    = 1
  triggers = { description = var.description }

  provisioner "local-exec" { command = "echo name " }
}

variable "description" {
  type = string
}

output "ids" {
  value = [for s in null_resource.name : s.id]
}
```
moduel2:
```hcl
provider "null" {}

resource "null_resource" "hi" {
  count    = length(var.resources)
  triggers = { t = "${var.resources[count.index]}" }

  provisioner "local-exec" {
    command = "echo greattings ${var.resources[count.index]}"
  }
}

variable "resources" {
  type = list(string)
}
```

module3:
```hcl
module "r" {
  source      = "../module1"
  description = var.description
}

module "hi" {
  source    = "../module2"
  resources = module.r.ids
}

variable "description" {
  type = string
}
```
### Debug Output

https://gist.github.com/eddytrex/43c5bd47b273607705ae0f9bf1931503

### Crash Output

```bash
Error: Cycle: module.r.output.ids, module.hi.var.resources, module.hi.null_resource.hi (prepare state), module.hi.null_resource.hi[0] (destroy), module.r.null_resource.name[0] (destroy), module.r.null_resource.name[0]
```

### Expected Behavior

Renew the null_resource in the module 1 
pass the values to the module 2  
then Renew null_resource in module 2

### Actual Behavior

Error Cycle

### Steps to Reproduce
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply -var description=first module3`
3. `terraform apply -var description=second module3`

### Additional Context
tested with terraform 0.12.6 with the same behavior

**Comment by eddytrex** [comment]
Using the lifecycle crate before destroy on the null_resource name, it work fine. But maybe is not positive to use create before destroy in other resources.

**Comment by teamterraform** [comment]
Thanks for the added info @eddytrex, that might be very useful in tracking this down.
You are correct that you should not need to use `create_before_destroy` to fix a cycle like this, it just coincidentally happens to break the cycle when destroying the instances.

**Comment by eddytrex** [comment]
I'm testing in 0.12.7, interesting if there si not count/each in the null_resource.hi in the module 2 that depends on the variable(resources) it works with out cycle.

**Comment by jbardin** [comment]
Closed by #22976
